ui/app/pages/confirm-transaction/conf-tx.js

```diff
@@ -45,188 +40,220 @@ function mapStateToProps (state) {
   }
 }
 
-inherits(ConfirmTxScreen, Component)
-function ConfirmTxScreen () {
-  Component.call(this)
-}
-
-ConfirmTxScreen.prototype.getUnapprovedMessagesTotal = function () {
-  const {
-    unapprovedMsgCount = 0,
-    unapprovedPersonalMsgCount = 0,
-    unapprovedTypedMessagesCount = 0,
-  } = this.props
-
-  return unapprovedTypedMessagesCount + unapprovedMsgCount + unapprovedPersonalMsgCount
-}
-
-ConfirmTxScreen.prototype.componentDidMount = function () {
-  const {
-    unapprovedTxs = {},
-    network,
-    send,
-  } = this.props
-  const unconfTxList = txHelper(unapprovedTxs, {}, {}, {}, network)
-
-  if (unconfTxList.length === 0 && !send.to && this.getUnapprovedMessagesTotal() === 0) {
-    this.props.history.push(DEFAULT_ROUTE)
+class ConfirmTxScreen extends Component {
+  static propTypes = {
+    unapprovedMsgCount: PropTypes.number,
+    unapprovedPersonalMsgCount: PropTypes.number,
+    unapprovedTypedMessagesCount: PropTypes.number,
+    network: PropTypes.number,
+    index: PropTypes.number,
+    unapprovedTxs: PropTypes.array,
+    unapprovedMsgs: PropTypes.array,
+    unapprovedPersonalMsgs: PropTypes.array,
+    unapprovedTypedMessages: PropTypes.array,
+    match: PropTypes.shape({
+      params: PropTypes.shape({
+        id: PropTypes.number,
+      }),
+    }),
+
+    selectedAddressTxList: PropTypes.array,
+    currentCurrency: PropTypes.string,
+    blockGasLimit: PropTypes.string,
+    conversionRate: PropTypes.number,
+    history: PropTypes.object,
+    accounts: PropTypes.array,
+    selectedAddress: PropTypes.string,
+    identities: PropTypes.object,
+    dispatch: PropTypes.func.isRequired,
+    send: PropTypes.shape({
+      to: PropTypes.string,
```

Comment:
.... strangely enough, this property is never set (in `ducks/send.duck.js`) :thinking: 

It's only used in this component to check whether it's falsy, so... that condition can be omitted really; it's always unset.

Reply:
I'm not sure this comes from there? This value _is_ set via `UPDATE_SEND_TO`, so `state.metamask.to` could be truthy sometimes? I'm going to leave this as-is for now since it's not immediately clear.

Reply:
Oops, yes you're correct, my bad. I thought it was from `state.send`, not `state.metamask.send`.